File: ClientApp/src/app/data/register-user.model.ts
export interface IRegisterUserData {
    FirstName: string;
    LastName: string;
    UserName: string;
    EmailAddress: string;
}

export class RegisterUserData implements IRegisterUserData {
    constructor(firstname, lastname, username, emailAddress: string) {
        this.FirstName = firstname;
        this.LastName = lastname;
        this.UserName = username;
        this.EmailAddress = emailAddress;
    }

    public FirstName: string;
    public LastName: string;
    public UserName: string;
    public EmailAddress: string;
}

File: Models/Account/RegisterUserModel.cs
using System.ComponentModel.DataAnnotations;

namespace Recepten.Models.Account
{
    public class RegisterUserModel
    {
        // Parameterless constructor needed for the framework.
        public RegisterUserModel() { }

        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [Required]
        public string UserName { get; set; }

        [Required]
        public string EMailAddress { get; set; }
    }
}